### PR TITLE
Blaze: Update campaign details after dismiss the detail view

### LINKED ISSUE #13795: Blaze: When a campaign is canceled, the status is not immediately updated.
To repro:

1. Go to a Product, then **Promote with Blaze**,
2. Create the campaign,
3. Go to **Menu > Blaze**, confirm that the campaign is listed with "In Moderation" status,
4. Tap the campaign to show **Campaign Details** sheet,
5. Scroll down, tap **Cancel Campaign**,
6. Confirm the cancellation,
7. Tap "Done" in the sheet,
8. Blaze campaigns list is now shown.
9. Note that the campaign status is still "In Moderation". **Expected status: Canceled**.

### PR DESCRIPTION
**Pull request by itsmeichigo**
Closes: #13795 

## Description

There's an issue with campaign details not updated on the dashboard and campaign list after canceling the campaign on the detail page (web view). Since there is no URL triggered when a campaign is canceled, we cannot detect when the status is updated.

This PR integrates the Blaze campaign detail retrieval endpoint and triggers it when dismissing the campaign detail screen. This ensures that the campaign status is updated correctly in case it's canceled on the web view.

## Steps to reproduce

- Log in to a test store eligible for Blaze.
- Create a new Blaze campaign then open its detail page from the Blaze dashboard card.
- Cancel the campaign and dismiss the detail view.
- Confirm that the campaign status is updated afterward without the need to refresh the dashboard.
- Repeat the same steps with the campaign list and confirm that campaign status is also updated after canceling a campaign from this screen.

## Testing information

Tested on simulator iOS 17.4 and confirmed that refreshing campaign details works correctly on both Blaze dashboard card and campaign list.

## Screenshots

https://github.com/user-attachments/assets/247fafde-772f-4ccf-9065-7e663a38d4f0

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/ViewRelated/Blaze/BlazeCampaignList/BlazeCampaignItemView.swift [hunk 1]
```diff
@@ -102,6 +102,7 @@ struct BlazeCampaignItemView: View {
                 .stroke(Color(uiColor: .separator), lineWidth: Layout.strokeWidth)
         }
         .padding(Layout.strokeWidth)
+        .contentShape(Rectangle())
```
Comment (itsmeichigo) [inline WooCommerce/Classes/ViewRelated/Blaze/BlazeCampaignList/BlazeCampaignItemView.swift:105]: This fixes the issue when tapping on the campaign item doesn't work consistently.

File: WooCommerce/Classes/ViewRelated/Dashboard/Blaze/BlazeCampaignDashboardViewModel.swift [hunk 2]
```diff
@@ -195,6 +199,17 @@ final class BlazeCampaignDashboardViewModel: ObservableObject {
             await reload()
         }
     }
+
+    func refreshSelectedCampaign() {
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Dashboard/Blaze/BlazeCampaignDashboardViewModel.swift:212]: Non-blocking suggestions: 
1. Consider adding a comment to explain how refreshing the campaign upon dismiss helps. 
2. We could also add a separate method `onDismiss` and call the `refreshSelectedCampaign` from there. This helps make the view dumb and keeps the reasoning inside the view model.

**Review by selanthiraiyan — APPROVED** [review]
Thanks for working on this, Huong! 

I tested by cancelling Blaze campaigns from webview and ensured that both dashboard and campaign list view are reloaded as expected. 

🚀

File: WooCommerce/Classes/ViewRelated/Dashboard/Blaze/BlazeCampaignDashboardViewModel.swift (hunk 2 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Dashboard/Blaze/BlazeCampaignDashboardViewModel.swift:212]: Done in 92b055d225963e56038b0bc2655300af0bb57f25.

**Pull request merged**